errors.go

```diff
@@ -0,0 +1,56 @@
+package filclient
+
+import "fmt"
+
+type ErrorCode int
+
+const (
+	// Failed to connect to a miner.
+	ErrMinerConnectionFailed ErrorCode = iota
+
+	// There was an issue related to the Lotus API.
+	ErrLotusError
+)
+
+type Error struct {
+	Code  ErrorCode
+	Inner error
+}
+
+func (code ErrorCode) String() string {
+	var msg string
+
+	switch code {
+	case ErrMinerConnectionFailed:
+		msg = "miner connection failed"
+	case ErrLotusError:
+		msg = "lotus error"
+	default:
+		msg = "(invalid error code)"
+	}
+
+	return msg
+}
+
+func (err *Error) Error() string {
+	return fmt.Sprintf("%v: %v", err.Code.String(), err.Inner.Error())
```

Comment:
```suggestion
	return fmt.Sprintf("%s: %s", err.Code, err.Inner)
```

Reply:
I wouldn't have thought about doing it this way, thanks!